### PR TITLE
CoreDNS application for Graduated project

### PR DESCRIPTION
**Pull request by fturib**
This a proposal for CoreDNS graduating as a "Graduated" projects hosted by CNCF.

### CODE REVIEW

**Comment by caniszczyk** [comment]
RFC @cncf/toc

File: reviews/graduation-coredns.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+
+# CoreDNS Graduation Application
+
+CoreDNS joined CNCF in March 2017 as inception project and graduated as incubation project on February 2018
+
+It has grown significantly over time either as a DNS Server available in Cloud Native environment, either as a key component of Kubernetes project:
+- In Fall 2017, CoreDNS was proposed to become a DNS Discovery Service for Kubernetes.
+- First version as alpha feature was released in v1.09 of Kubernetes - Dec 2017
```
Comment (rajansandeep) [inline reviews/graduation-coredns.md]:
```suggestion
- First version as alpha feature was released in v1.9 of Kubernetes - Dec 2017
```

File: reviews/graduation-coredns.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+### Document that it is being used successfully in production by at least three independent end users which, in the TOC’s judgement, are of adequate quality and scope.
+
+* Bose, SoundCloud, Infoblox, Zalando, Tradeshift, Yandex, AdGuard are knows users of CoreDNS. More adopters are listed [here](https://github.com/coredns/coredns/blob/master/ADOPTERS.md).
+
+* On Kubernetes side, from [a recent CNCF Survey](https://docs.google.com/forms/d/e/1FAIpQLSehsvd0n2HKnPEifukNHej4gMBrXe2saJOmxAmPq34dTGMWKA/viewanalytics), 50% deployments of cluster are using CoreDNS as the DNS discovery service
+
+
+### Have a healthy number of committers. A committer is defined as someone with the commit bit; i.e., someone who can accept contributions to some or all of the project.
+
+CoreDNS has currenlty 16 maintainers.
```
Comment (rajansandeep) [inline reviews/graduation-coredns.md]:
```suggestion
CoreDNS currently has 16 maintainers.
```

File: reviews/graduation-coredns.md (hunk 1 above)
Comment (fturib) [reply reviews/graduation-coredns.md]: ok

File: reviews/graduation-coredns.md (hunk 2 above)
Comment (fturib) [reply reviews/graduation-coredns.md]: ok

**Comment by caniszczyk** [comment]
RFC @cncf/toc 

presentation from CoreDNS team: https://docs.google.com/presentation/d/1riZ5g9NGBLOSfTMY3PGZX3ikSvxuihgbZzZp53bqWec/edit?usp=sharing

File: reviews/graduation-coredns.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+
+# CoreDNS Graduation Application
+
+CoreDNS joined CNCF in March 2017 as inception project and graduated as incubation project on February 2018
+
+It has grown significantly over time either as a DNS Server available in Cloud Native environment, either as a key component of Kubernetes project:
```
Comment (jonboulle) [inline reviews/graduation-coredns.md]: It has grown significantly over time in maturity and user base, both as a DNS Server available in Cloud Native environment and as a key component of Kubernetes project:

File: reviews/graduation-coredns.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+- First version as alpha feature was released in v1.9 of Kubernetes - Dec 2017
+- CoreDNS is considered the “default” DNS Discovery Service for Kubernetes v1.13 - to be released in Dec 2018
+
+The following application links to the required information to become a graduated project.
+
+## CoreDNS fulfills all the incubating and graduation criteria:
+
+### Document that it is being used successfully in production by at least three independent end users which, in the TOC’s judgement, are of adequate quality and scope.
+
+* Bose, SoundCloud, Infoblox, Zalando, Tradeshift, Yandex, AdGuard are knows users of CoreDNS. More adopters are listed [here](https://github.com/coredns/coredns/blob/master/ADOPTERS.md).
```
Comment (jonboulle) [inline reviews/graduation-coredns.md]: More adopters and details of their use

File: reviews/graduation-coredns.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+They are listed as ‘approvers’ on our [OWNER document](https://github.com/coredns/coredns/blob/master/OWNERS)
+
+### Demonstrate a substantial ongoing flow of commits and merged contributions.
+
+More than [100+ contributors](https://github.com/coredns/coredns/graphs/contributors) and 12 [releases](https://coredns.devstats.cncf.io/d/3/community-stats?orgId=1&var-period=d&var-repo_name=CoreDNS) since incubation graduation
+
+### Have committers from at least two organizations.
+
+We currently have [16 maintainers](https://github.com/coredns/coredns/blob/master/OWNERS).
+Most of them are contributing on their spare time and each considered at its own organization,
```
Comment (jonboulle) [inline reviews/graduation-coredns.md]: Most of them are contributing in their spare time; but, considering organizations, 4 maintainers are working for 2 distinct companies

File: reviews/graduation-coredns.md (hunk 3 above)
Comment (fturib) [reply reviews/graduation-coredns.md]: ok

File: reviews/graduation-coredns.md (hunk 5 above)
Comment (fturib) [reply reviews/graduation-coredns.md]: ok

File: reviews/graduation-coredns.md (hunk 4 above)
Comment (fturib) [reply reviews/graduation-coredns.md]: let me think how I can get some information here...

**Comment by caniszczyk** [comment]
project health dashboard:  https://all.devstats.cncf.io/d/54/project-health?orgId=1&var-repogroup_name=CoreDNS

**Comment by caniszczyk** [comment]
CoreDNS graduation vote out: https://lists.cncf.io/g/cncf-toc/message/2762

**Comment by caniszczyk** [comment]
https://www.cncf.io/announcement/2019/01/24/coredns-graduation/

+1 binding TOC votes (7/9):
Alexis: https://lists.cncf.io/g/cncf-toc/message/2776
Quinton: https://lists.cncf.io/g/cncf-toc/message/2786
Jon: https://lists.cncf.io/g/cncf-toc/message/2788
Camille: https://lists.cncf.io/g/cncf-toc/message/2799
Bryan: https://lists.cncf.io/g/cncf-toc/message/2800
Ken: https://lists.cncf.io/g/cncf-toc/message/2810
Ben: https://lists.cncf.io/g/cncf-toc/message/2811

**Pull request merged**